From 6917f035f8f391de15b39e610b4eb389489a2781 Mon Sep 17 00:00:00 2001
From: hienhuynh <hien.huynh.px@renesas.com>
Date: Tue, 23 Jun 2020 10:41:32 +0700
Subject: [PATCH 2/2] plat: rcar: timer: bl2_swdt: increase WDT count when ECC
 enabled

When enable ECC, access to dram have lower speed, sometimes Watchdog
timer get overflow when transfering Uboot binary from flash to DRAM.
This commit increase WDT counter to it's maximum value.

Signed-off-by: Binh Nguyen <binh.nguyen.uw@renesas.com>
Signed-off-by: hienhuynh <hien.huynh.px@renesas.com>
---
 drivers/renesas/rcar/watchdog/swdt.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/renesas/rcar/watchdog/swdt.c b/drivers/renesas/rcar/watchdog/swdt.c
index 111e651..7b94d3d 100644
--- a/drivers/renesas/rcar/watchdog/swdt.c
+++ b/drivers/renesas/rcar/watchdog/swdt.c
@@ -49,6 +49,12 @@ extern void gicd_set_icenabler(uintptr_t base, unsigned int id);
 #define WTCSRA_INIT_DATA		(WTCSRA_UPPER_BYTE + 0x0FU)
 #define WTCSRB_INIT_DATA		(WTCSRB_UPPER_BYTE + 0x21U)
 
+#if (RZG_DRAM_ECC_FULL == 1)
+#define WTCNT_COUNT_8p13k		(0U)
+#define WTCNT_COUNT_8p13k_H3VER10	(0U)
+#define WTCNT_COUNT_8p22k		(0U)
+#define WTCNT_COUNT_7p81k		(0U)
+#else //(RZG_DRAM_ECC_FULL == 1)
 #if RCAR_LSI == RCAR_D3
 #define WTCNT_COUNT_8p13k		(0x10000U - 40760U)
 #else
@@ -57,6 +63,7 @@ extern void gicd_set_icenabler(uintptr_t base, unsigned int id);
 #define WTCNT_COUNT_8p13k_H3VER10	(0x10000U - 20343U)
 #define WTCNT_COUNT_8p22k		(0x10000U - 41115U)
 #define WTCNT_COUNT_7p81k		(0x10000U - 39062U)
+#endif //(RZG_DRAM_ECC_FULL == 1)
 #define WTCSRA_CKS_DIV16		(0x00000002U)
 
 static void swdt_disable(void)
-- 
2.7.4

